File: zero-server/src/main/kotlin/com/zeroday/util/IdGenerator.kt
package com.zeroday.util

import java.util.UUID

object IdGenerator {
    fun newId(): String = UUID.randomUUID().toString()
    fun shortId(): String = UUID.randomUUID().toString().take(8)
    fun sessionId(playerId: String): String = "session_${playerId}_${System.currentTimeMillis()}"
    fun taskInstanceId(): String = "task_${shortId()}"
    fun eventInstanceId(): String = "evt_${shortId()}"
}

File: zero-server-csharp/src/ZeroDayMMO.Server/Util/IdGenerator.cs
namespace ZeroDayMMO.Server.Util;

public static class IdGenerator
{
    public static string NewId() => Guid.NewGuid().ToString();

    public static string ShortId() => Guid.NewGuid().ToString()[..8];

    public static string SessionId(string playerId) =>
        $"session_{playerId}_{DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()}";

    public static string TaskInstanceId() => $"task_{ShortId()}";

    public static string EventInstanceId() => $"evt_{ShortId()}";
}
